tests/php/sync/test_class.jetpack-sync-sso.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Testing Activation and Deactivation of Modules
+ */
+class WP_Test_Jetpack_Sync_SSO extends WP_Test_Jetpack_New_Sync_Base {
+	function test_sync_sso_is_two_step_required_filter_true() {
+		add_filter( 'jetpack_sso_require_two_step', '__return_true' );
+		$this->client->do_sync();
+		remove_filter( 'jetpack_sso_require_two_step', '__return_true' );
+
+		$callableValue = $this->server_replica_storage->get_callable( 'sso_is_two_step_required' );
+
+		$this->assertEquals( '1', $callableValue );
+	}
+
+	function test_sync_sso_should_hide_login_form_filter_true() {
+		add_filter( 'jetpack_remove_login_form', '__return_true' );
+		$this->client->do_sync();
+		remove_filter( 'jetpack_remove_login_form', '__return_true' );
+
+		$callableValue = $this->server_replica_storage->get_callable( 'sso_should_hide_login_form' );
+
+		$this->assertEquals( '1', $callableValue );
+	}
+
+	function test_sync_sso_match_by_email_filter_true() {
+		add_filter( 'jetpack_sso_match_by_email', '__return_true' );
+		$this->client->do_sync();
+		remove_filter( 'jetpack_sso_match_by_email', '__return_true' );
+
+		$callableValue = $this->server_replica_storage->get_callable( 'sso_match_by_email' );
+
+		$this->assertEquals( '1', $callableValue );
+	}
+
+	function test_sync_sso_new_user_override_filter_true() {
+		add_filter( 'jetpack_sso_new_user_override', '__return_true' );
+		$this->client->do_sync();
```

Comment:
I think it would make sense to set an assert here to make sure that the function is also returning true.